src/ChannelServer/Network/Handlers/Parties.cs

```diff
@@ -0,0 +1,308 @@
+﻿// Copyright (c) Aura development team - Licensed under GNU GPL
+// For more information, see license file in the main folder
+
+using Aura.Channel.Network.Sending;
+using Aura.Channel.Util;
+using Aura.Channel.World.Entities;
+using Aura.Data;
+using Aura.Data.Database;
+using Aura.Mabi.Const;
+using Aura.Mabi.Network;
+using Aura.Shared.Network;
+using System;
+
+namespace Aura.Channel.Network.Handlers
+{
+    public partial class ChannelServerHandlers : PacketHandlerManager<ChannelClient>
+    {
+        [PacketHandler(Op.PartyCreate)]
+        public void PartyCreate(ChannelClient client, Packet packet)
+        {
+            var creature = client.GetCreatureSafe(packet.Id);
+            
+            if (creature == null)
+                return;
+
+            if (creature.IsInParty)
+                return;
+
+            // Needs check like the below, but do homesteads count as dynamic regions? I know you can't make a party in the shadow realm (you can in dungeons), but are they the only places? Need to look into that.
+            //if (!creature.Region.IsDynamic)          
+            {
+                var party = creature.Party;
+
+                party.CreateParty(creature, packet);
+
+                Send.CreatePartyR(creature);
+
+                party.Open();
+                return;
+            }
+            //else
+            //Send.CreatePartyR(creature);
+        }
+
+        [PacketHandler(Op.PartyJoin)]
+        public void PartyJoin(ChannelClient client, Packet packet)
+        {
+            var creature = client.GetCreatureSafe(packet.Id);
+
+            if (creature == null)
+                return;
+
+            if (creature.IsInParty)
+                return;
+
+            var partyLeader = ChannelServer.Instance.World.GetCreature(packet.GetLong());
+
+            if (!partyLeader.IsInParty)
```

Comment:
Unless you checked that not sending any response doesn't lock the character you generally shouldn't just return. If it it's something that shouldn't happen unless the user cheated you should throw a security violation exception, if it's something that *might* happen, but *shouldn't* (and with devCAT everything is possible, seriously) you should send a fail response.